docs/language/functions.rst

```diff
@@ -279,6 +279,38 @@ constructor.
     for storage rent and there is a minimum deposit needed for the contract. As a result,
     constructors always receive value on Parity Substrate.
 
+Overriding function selector
+____________________________
+
+When a function is called, the function selector and the arguments to the function
+are serialized (also known as abi encoded) and passed to the program. The selector
+is what the runtime program uses to determine what function was called. Usually
+the function selector is generated from a deterministic hash value of the function
+name and its arguments.
+
+The selector value can be overriden with the ``selector=hex"deadbea1"`` syntax,
+for example:
+
+.. code-block:: solidity
+
+    contract foo {
+        function get_foo() selector=hex"01020304" public returns (int) {
+            return 102;
+        }
+    }
+
+Only ``public`` and ``external`` functions have a selector, and can have their
+selector overriden. On Substrate, constructors have selectors too, so they
+can also have their selector overriden. If a function overrides another one in a
+base contract, then the selector of both must match.
+
+.. warning::
+    On ewasm and Solana, changing the selector may result in a mismatch between
+    the contract metadata and the actual contract code, because the metadata does
+    not explicitly store the selector.
+
+    Only use this feature if you know what you are doing.
```

Comment:
Instead of saying `Only use this feature if you know what you are doing`, I think it is more polite to say `Use this feature carefully, as it may either break a contract or cause undefined behavior.` What do you think?

Reply:
I've added this, thanks